### DISCUSSION TITLE
cam pan v2 ptz control via blue iris

### DISCUSSION THREAD

**Discussion by vincom** [gtxaspec/wz_mini_hacks#696] [Q&A]
in blue iris ptz settings, which camera do i need to emulate the pan v2 camera, is it even possible?

**Comment by endertable** [comment]
1.  Out of the box, I believe Wyze uses it's stream data channel to receive commands, like the PTZ and many others.  In BI, you will not be able to use an existing template, but you can create your own since you are using wz_mini.  But you might first have to create a cgi web page on the cam(s) to run the "cmd move X Y" commands on the cam for you.  

2.  Another alternative, which will be more necessary if you want to control the V3 Pan, will be to use @mrlt8's docker_wyze_bridge of which you can find some directions here for BI:   https://github.com/mrlt8/docker-wyze-bridge/issues/728  
In a response, @mrlt8 states you can also use MQTT commands in a response.  These cams have several subs that use MQTT so I am not sure if he is referring to the stream channel, Wyze's cloud IOT, or the one for docker_wyze_bridge itself.

3.  If you don't want to use docker and want to keep things simpler to have just PTZ , including for the Pan V3, you can use https://github.com/kroo/wyzecam, but do note you will have to do some updates for it to work as it is old and unmaintained.  It needs information details of the cam(s) to authenticate to the cams to start a "stream", which is relatively easy to get even using Wyze's own API key system.  But that is just one part.  You will also have to create a web/CGI interface to the python daemon to issue the 'K' commands to the cam(s).

I believe I sorted them easiest to most difficult for setup, speaking loosely.  wz_mini, option 1, would not require another 'server' either, as you can issue CGI direct to cam if you use a web server on the cam.   If you have Pan V3 cams, then probably best not to use option 1 as you don't have a way to control those with wz_mini at this time.  Purchase a RPi and load docker_wyze_bridge up.  Actually you can load docker_wyze_bridge on anything you can load docker on.  Understand it is a full cam suite and PTZ control is an almost insignificant subset of what all it can do.  Option 3 is not for the faint of heart, unless you have CGI and python skills, then it will be your best option after you get it set up, especially if you want to be ready for Pan V3 cams and beyond; and if you got this far with it, adding other config commands is trivial at best, which would work for all Wyze cams.  It, also, will require to run on an RPi or some other MiTM server.

Good luck.
If you got yours set up already with BI, please share your findings...

**Comment by vincom** [comment]
thx, i just finishing installing and configuring mrlt8 wyze bridge in docker desktop. took awhile to figure on how to use docker and then configure blue iris video urls, but got it going now. with mrlt8 i can now use wyze door bell and pan v3 in blue iris, which is very cool and the video feed seems very stable.
im a noob to docker stuff so took awhile to figure it all out.
thnks for the links, i will read it and see if i can make heads or tails out of it for the ptz

![blueirisdocker](https://github.com/gtxaspec/wz_mini_hacks/assets/2191155/f18d510d-a76b-4250-a4c2-caaaa084c4bd)

**Reply by endertable** [reply]
Great job, you got it up and running pretty fast.  Hopefully that link helped you with the PTZ command.  I was looking through some of my old code when I tested that soft and I used:

/api/cam_name/rotary_degree/X,Y

 X goes right
-X  goes left
 Y goes up
-Y goes down

Something to try if others don't work...
